src/cargo/core/compiler/mod.rs

```diff
@@ -439,6 +455,7 @@ fn link_targets<'a, 'cfg>(
                 profile: art_profile,
                 features,
                 filenames: destinations,
+                executable: executable,
```

Comment:
Could be just `executable`: if field name and var name are the same, you don't have to write `foo: foo`.